Building on Intel Raptor Lake fails

I would say it's an issue detecting the CPU features as forcing the TARGET=HAWELL make it work? Thanks!

Ubuntu 22.04, on Intel i5-13500, with gcc-11:
```bash
make -j
...
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:99,
                 from /root/OpenBLAS/kernel/x86_64/../generic/../simd/intrin.h:51,
                 from /root/OpenBLAS/kernel/x86_64/../generic/dot.c:30,
                 from /root/OpenBLAS/build/kernel/CMakeFiles/sdot_k.c:7:
/usr/lib/gcc/x86_64-linux-gnu/11/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
   63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
      | ^~~~~~~~~~~~~~~
In file included from /root/OpenBLAS/kernel/x86_64/../generic/../simd/intrin.h:63,
                 from /root/OpenBLAS/kernel/x86_64/../generic/dot.c:30,
                 from /root/OpenBLAS/build/kernel/CMakeFiles/sdot_k.c:7:
/root/OpenBLAS/kernel/x86_64/../generic/../simd/intrin_avx.h:22:26: note: called from here
   22 |     #define v_muladd_f32 _mm256_fmadd_ps
/root/OpenBLAS/kernel/x86_64/../generic/dot.c:79:21: note: in expansion of macro ‘v_muladd_f32’
   79 |             vsum0 = v_muladd_f32(
      |                     ^~~~~~~~~~~~
make[2]: *** [kernel/CMakeFiles/kernel.dir/build.make:341: kernel/CMakeFiles/kernel.dir/CMakeFiles/sdot_k.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:584: kernel/CMakeFiles/kernel.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

getarch 1:
```
#define INTEL_UNKNOWN
#define L1_CODE_SIZE 32768
#define L1_CODE_ASSOCIATIVE 8
#define L1_CODE_LINESIZE 64
#define L1_DATA_SIZE 49152
#define L1_DATA_ASSOCIATIVE 12
#define L1_DATA_LINESIZE 64
#define L2_SIZE 1310720
#define L2_ASSOCIATIVE 7
#define L2_LINESIZE 64
#define DTB_DEFAULT_ENTRIES 32
#define HAVE_CMOV
#define HAVE_MMX
#define HAVE_SSE
#define HAVE_SSE2
#define HAVE_SSE3
#define HAVE_SSSE3
#define HAVE_SSE4_1
#define HAVE_SSE4_2
#define HAVE_AVX
#define HAVE_AVX2
#define HAVE_FMA3
#define HAVE_CFLUSH
#define NUM_SHAREDCACHE 2
#define NUM_CORES 64
#define CORE_PRESCOTT
```

--- a/cpuid_x86.c
+++ b/cpuid_x86.c
@@ -1547,6 +1547,8 @@ int get_cpuname(void){
       case 11: //family 6 exmodel 11
         switch (model) {
           case 7: // Raptor Lake
+          case 10:
+          case 15:
             if(support_avx2())
               return CPUTYPE_HASWELL;
 	    if(support_avx())
@@ -2348,6 +2350,8 @@ int get_coretype(void){
       case 11:
 	switch (model) {
 	  case 7: // Raptor Lake
+          case 10:
+          case 15:
 #ifndef NO_AVX2
 	  if(support_avx2())
             return CORE_HASWELL;

